smartswitchplugin/src/main/java/com/kaltura/playkit/plugins/smartswitch/SmartSwitchExecutor.kt

```diff
@@ -0,0 +1,180 @@
+package com.kaltura.playkit.plugins.smartswitch
+
+import android.net.Uri
+import androidx.annotation.Nullable
+import com.google.gson.Gson
+import com.kaltura.playkit.PKLog
+import com.kaltura.playkit.plugins.smartswitch.pluginconfig.CDNList
+import com.kaltura.playkit.plugins.smartswitch.pluginconfig.SmartSwitchErrorResponse
+import com.kaltura.playkit.plugins.smartswitch.pluginconfig.SmartSwitchParser
+import java.io.BufferedReader
+import java.io.IOException
+import java.io.InputStream
+import java.io.InputStreamReader
+import java.net.HttpURLConnection
+import java.net.MalformedURLException
+import java.net.URL
+import java.util.concurrent.Callable
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.concurrent.Future
+
+internal class SmartSwitchExecutor {
+
+    private val smartSwitchExecutor: ExecutorService = Executors.newSingleThreadExecutor()
+
+    fun sendRequestToYoubora(accountCode: String, originCode: String,
+                             resourceUrl: String?,
+                             @Nullable optionalParams: HashMap<String, String>?,
+                             @Nullable smartSwitchUrl: String): Future<Pair<String, String>?>? {
+        val sendConfigToYoubora = SendConfigToYoubora(smartSwitchUrl, accountCode, originCode, resourceUrl, optionalParams)
+        return smartSwitchExecutor.submit(sendConfigToYoubora)
+    }
+
+    fun terminateService() {
+        smartSwitchExecutor.shutdownNow()
+    }
+
+    /**
+     * Callable which is handling the Network request to CDN Balancer
+     * Does the parsing of the response
+     * Sends the callback as well
+     */
+    private class SendConfigToYoubora(val smartSwitchServerUrl: String,
+                                      val accountCode: String,
+                                      val originCode: String,
+                                      var resourceUrl: String?,
+                                      val optionalParams: HashMap<String, String>?): Callable<Pair<String, String>?> {
+
+        private val log: PKLog = PKLog.get("SmartSwitchExecutor")
+
+        private val connectionReadTimeOut: Int = 10000
+        private val connectionTimeOut: Int = 10000
+        private val successResponseCode: Int = 200
+        private val requestMethod: String = "GET"
+        private val accountCodeKey = "accountCode"
+        private val resourceKey = "resource"
+        private val originCodeKey = "originCode"
+        private var errorMessage = "Invalid Response"
+
+        override fun call(): Pair<String, String> {
+            var connection: HttpURLConnection? = null
+            val inputStream: InputStream?
+            var smartSwitchUri: Uri
+            try {
+                smartSwitchUri = Uri.parse(smartSwitchServerUrl)
```

Comment:
maybe call this  in 1 line?

smartSwitchUri = appendQueryParams(Uri.parse(smartSwitchServerUrl))

Reply:
Done